test/TestNamespaceTracker.hs

```diff
@@ -246,37 +246,18 @@ testSubscribeAsClient =
     events = [
         ClientData alice [msg ["hello"] AssignType],
         ClientData bob [msg ["hello"] Subscribe],
+        ClientData alice [msg ["hello"] Add],
         ClientDisconnect alice
       ]
-    -- forTest = do
-    --   sendFwd $ ClientData alice [msg ["hello"] AssignType]
-    --   sendFwd $ ClientData bob [msg ["hello"] Subscribe]
-    --   -- sendFwd $ ClientData charlie [msg ["hello"] Subscribe]
-    --   -- sendFwd $ ClientData dave [
-    --   --   msg ["hello"] Subscribe, msg ["hello"] Unsubscribe]
-    --   -- sendFwd $ ClientData ethel [msg ["hello"] Subscribe]
-    --   -- sendFwd $ ClientData alice [msg ["hello"] Set]
-    --   sendFwd $ ClientDisconnect alice
-    --   resps <- collectAllResponsesUntil alice
-    --   lift $ assertEqual "poop" mempty resps
-    --   undefined
   in do
     resps <- gogo events alice protocol
-    assertEqual "resps" [] resps
---     response <- trackerHelper [
---         (42, Alice, [msg ["hello"] AssignType]),
---         (43, Alice, [msg ["hello"] Subscribe]),
---         (44, Alice, [msg ["hello"] Subscribe]),
---         (45, Alice, [msg ["hello"] Subscribe, msg ["hello"] Unsubscribe]),
---         (46, Alice, [msg ["hello"] Subscribe]),
---         (46, Alice, [msg ["hello"] Unsubscribe]),
---         (42, Alice, [msg ["hello"] Set])]
---     assertEqual "response check" (Map.fromList [
---         (43, [[msg ["hello"] Remove], [msg ["hello"] Set]]),
---         (44, [[msg ["hello"] Remove], [msg ["hello"] Set]]),
---         (46, [[msg ["hello"] Remove]])
---         ])
---         response
+    assertEqual "resps" [
+        -- Because the fake server bounces everything you get subscribes
```

Comment:
Is this a FIXME? Should we effectively be actually trying to test a bit more of the integrated behaviour, or is the isolated test useful?

Reply:
Don't know, the isolated ones did find some things.

Reply:
Cool - I guess we leave it as-is and then come back to it if it feels like it's got holes...